module/actor/actor.js

```diff
@@ -23,30 +23,25 @@ export class yzecoriolisActor extends Actor {
     //setup default images for ships
     if (
       initData.type === "ship" &&
-      hasProperty(initData, "img") &&
-      initData.img === Actor.DEFAULT_ICON
+      ((hasProperty(initData, "img") && initData.img === Actor.DEFAULT_ICON) ||
+        !hasProperty(initData, "img"))
     ) {
       this.updateSource({ img: CONFIG.YZECORIOLIS.DEFAULT_SHIP_KEY_ART });
     }
 
     // we check the incoming data to make sure we aren't overriding a 'cloning'
     // operation.
     if (
-      !hasProperty(initData, "system.keyArt") &&
+      !hasProperty(initData, "img") &&
       (initData.type === "character" || initData.type === "npc")
     ) {
       this.updateSource({
-        "system.keyArt": CONFIG.YZECORIOLIS.DEFAULT_PLAYER_KEY_ART,
-      });
-    }
-    // if we have a blank string keyArt prop, just set it to the default.
-    if (
-      hasProperty(initData, "system.keyArt") &&
-      (initData.type === "character" || initData.type === "npc") &&
-      initData.system.keyArt === ""
-    ) {
-      this.updateSource({
-        "system.keyArt": CONFIG.YZECORIOLIS.DEFAULT_PLAYER_KEY_ART,
+        img: CONFIG.YZECORIOLIS.DEFAULT_PLAYER_KEY_ART,
+        prototypeToken: {
+          texture: {
+            src: CONFIG.YZECORIOLIS.DEFAULT_PLAYER_KEY_ART_TOKEN,
```

Comment:
Set default token art otherwise Foundry will default to copy the img field, which leads to a not interesting token.